### PR TITLE
[FEATURE-UIPT-60] - Product feedbacks

### PR DESCRIPTION
**Pull request by IrynaMukhina**

### CODE REVIEW

File: src/app/components/product-review/product-review.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      size="24px"
+      readonly="false"
+      [value]="0"
+      (rate)="onRate($event)">
+    </star-rating>
+    <textarea #feedback
+      class="feedback__text"
+      maxLength="{{ maxCharNum }}"
+      (keyup)="charCount($event)">
+    </textarea>
```
Comment (castan1c) [inline src/app/components/product-review/product-review.html]: `></textarea>` Align it like this and watch the trick with 4 spaces removed
" everything between <textarea> and </textarea> is used as the default value for your textarea box."

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "@fortawesome/free-solid-svg-icons": "^5.11.2",
     "@ngrx/effects": "^8.3.0",
     "@ngrx/store": "^8.3.0",
+    "ng-starrating": "^1.0.7",
```
Comment (TorskyM) [inline package.json]: Any specific reason why we can't use `ngx-star-rating` instead?
We supposed to use `ngx` packages

File: src/app/components/product-review/product-review.component.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import { Component, ViewChild, Input, OnInit, ElementRef, OnDestroy } from '@angular/core';
```
Comment (TorskyM) [inline src/app/components/product-review/product-review.component.ts]: Import align, please

File: src/app/components/product-review/product-review.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    <span class="characters_left">({{ charLeft }})</span>
+    <app-button (click)="sendFeedback()">Leave Feedback</app-button>
+  </div>
+  <h3 class="feedback__title">Custom feedbacks</h3>
+  <div class="feedback__review">
+    <div *ngIf="currentProductReviews$ | async as reviews">
+      <div *ngIf="reviews?.length; else noReviews">
+        <div class="feedback" *ngFor="let review of reviews">
```
Comment (TorskyM) [inline src/app/components/product-review/product-review.html]: Atribut ordering)

File: src/app/components/product-review/product-review.scss [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  &__details {
+    display: flex;
+    align-items: flex-start;
+  }
+
+  &__avatar {
+    margin: 10px 10px 10px 0;
+    height: 50px;
+    border-radius: 50px;
+    box-shadow: 0 0 0 3px $accentColor, 0 0 13px $bgColorMedium;
```
Comment (TorskyM) [inline src/app/components/product-review/product-review.scss]: `margin
box-shadow
border
height`

File: src/app/app.enum.ts [hunk 6]
```diff
@@ -31,5 +32,7 @@ export enum ToastrMessage {
   invalidModal = 'modal must have an id',
   loginFailed = 'Incorrect email and/or password.',
   signUpFailed = 'That email is already in use.',
-  accessDenied = 'To see cart page, login first'
+  accessDenied = 'To see cart page, login first',
+  successfulFeedback = 'Your feedback has been sent successfully',
+  uncorrectFeedback = 'Please enter message or rate product'
```
Comment (andriiveremchuk) [inline src/app/app.enum.ts]: *incorrect

File: src/app/components/product-review/product-review.component.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  public ngOnInit(): void {
+    this.charLeft = this.maxCharNum;
+
+    this.getProductReview();
+  }
+
+  public charCount(): void {
+    const enteredCharNum = this.feedbackField.nativeElement.value.length;
+
+    if ( this.maxCharNum - enteredCharNum >= 0) {
```
Comment (andriiveremchuk) [inline src/app/components/product-review/product-review.component.ts]: is it possible to move this expression to variable?

File: src/app/components/product-review/product-review.component.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    this.clearTextArea();
+  }
+
+  public onRate($event: {
+    oldValue: number,
+    newValue: number,
+    starRating: StarRatingComponent
+  }): void {
+    this.userReview.rating = $event.newValue;
+  }
```
[the remark this answers is not in the document]
Comment (IrynaMukhina) [reply src/app/components/product-review/product-review.component.ts]: I think that with object it's more visible from where we get 'newValue', so I preferred to leave my version.

File: package.json (hunk 2 above)
Comment (IrynaMukhina) [reply package.json]: any reason to don't use ng-starrating?

File: src/app/components/product-review/product-review.component.ts (hunk 3 above)
Comment (IrynaMukhina) [reply src/app/components/product-review/product-review.component.ts]: It's within the allowed char limit

**Comment by IrynaMukhina** [comment]
@TorskyM @andriiveremchuk @castan1c thank you very much for reviewing

**Pull request merged**